packages/lesswrong/server/collections/reports/mutations.ts

```diff
@@ -10,8 +10,46 @@ import { makeGqlCreateMutation, makeGqlUpdateMutation } from "@/server/vulcan-li
 import { getLegacyCreateCallbackProps, getLegacyUpdateCallbackProps, insertAndReturnCreateAfterProps, runFieldOnCreateCallbacks, runFieldOnUpdateCallbacks, updateAndReturnDocument, assignUserIdToData } from "@/server/vulcan-lib/mutators";
 import gql from "graphql-tag";
 import cloneDeep from "lodash/cloneDeep";
+import { ChatPostMessageResponse, WebClient } from "@slack/web-api";
 
 
+async function postReportsToSunshine(report: DbReport, context: ResolverContext): Promise<ChatPostMessageResponse | undefined> {
+
+  const slackBotToken = process.env.AMANUENSIS_SLACK_BOT_TOKEN;
+  if (!slackBotToken) {
+    return;
+  }
+
+  const user = await context.Users.findOne({ _id: report.userId });
+  const [comment, post] = await Promise.all([
+    report.commentId ? context.Comments.findOne({ _id: report.commentId }) : null,
+    report.postId ? context.Posts.findOne({ _id: report.postId }) : null,
+  ]);
+  const reportedUser = await context.Users.findOne({ _id: report.reportedUserId ?? comment?.userId ?? post?.userId });
+
+  const slack = new WebClient(slackBotToken);
+
+  return await slack.chat.postMessage({
+    channel: 'C3GTABBTQ',
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `${user?.displayName ?? report.userId} reported <https://lesswrong.com${report.link}|this ${report.commentId ? 'comment' : report.postId ? 'post' : report.reportedUserId ? 'user' : ''}> by ${reportedUser?.displayName ?? report.reportedUserId}`,
```

Comment:
something about this text looks malformed: `<https://lesswrong.com${report.link}|this`